frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/OptionListEditor/OptionListEditor.tsx

```diff
@@ -1,184 +1,69 @@
-import React, { createRef, useRef, useState } from 'react';
-import type { Option } from 'app-shared/types/Option';
+import React, { forwardRef } from 'react';
 import { useTranslation } from 'react-i18next';
-import {
-  StudioCodeListEditor,
-  StudioModal,
-  StudioSpinner,
-  StudioErrorMessage,
-  StudioAlert,
-  StudioProperty,
-} from '@studio/components';
-import type { CodeListEditorTexts } from '@studio/components';
+import { StudioSpinner, StudioErrorMessage, StudioAlert } from '@studio/components';
 import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
-import { useUpdateOptionListMutation } from 'app-shared/hooks/mutations/useUpdateOptionListMutation';
 import { useOptionListsQuery } from 'app-shared/hooks/queries/useOptionListsQuery';
-import { useOptionListEditorTexts } from '../../hooks/useOptionListEditorTexts';
-import { usePreviewContext } from 'app-development/contexts/PreviewContext';
 import type { IGenericEditComponent } from '../../../../../componentConfig';
 import type { SelectionComponentType } from '../../../../../../../types/FormComponent';
-import classes from './OptionListEditor.module.css';
+import { LibraryOptionsEditor } from './LibraryOptionsEditor';
+import { ManualOptionsEditor } from './ManualOptionsEditor';
+import { handleOptionsIdChange } from '../utils/utils';
 
-type OptionListEditorProps = {
-  optionsId: string;
-  label: string;
-} & Pick<IGenericEditComponent<SelectionComponentType>, 'component' | 'handleComponentChange'>;
+export type OptionListEditorProps = Pick<
+  IGenericEditComponent<SelectionComponentType>,
+  'component' | 'handleComponentChange'
+>;
 
-export function OptionListEditor({
-  optionsId,
-  component,
-  handleComponentChange,
-  label,
-}: OptionListEditorProps): React.ReactNode {
-  const { t } = useTranslation();
-  const { org, app } = useStudioEnvironmentParams();
-  const { data: optionsLists, status } = useOptionListsQuery(org, app);
+export const OptionListEditor = forwardRef<HTMLDialogElement, OptionListEditorProps>(
+  ({ component, handleComponentChange }: OptionListEditorProps, dialogRef): React.ReactNode => {
+    const { t } = useTranslation();
+    const { org, app } = useStudioEnvironmentParams();
+    const { data: optionsLists, status } = useOptionListsQuery(org, app);
 
-  switch (status) {
-    case 'pending':
-      return (
-        <StudioSpinner spinnerTitle={t('ux_editor.modal_properties_code_list_spinner_title')} />
-      );
-    case 'error':
+    const handleDelete = () => {
+      handleOptionsIdChange({ component, handleComponentChange, optionsId: undefined });
```

Comment:
Same comment as above; we should create a new object to the handle-function instead of altering the current.